test/e2erp/compute.go

```diff
@@ -0,0 +1,203 @@
+package e2erp
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	fakepubkey = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC7laRyN4B3YZmVrDEZLZoIuUA72pQ0DpGuZBZWykCofIfCPrFZAJgFvonKGgKJl6FGKIunkZL9Us/mV4ZPkZhBlE7uX83AAf5i9Q8FmKpotzmaxN10/1mcnEE7pFvLoSkwqrQSkrrgSm8zaJ3g91giXSbtqvSIj/vk2f05stYmLfhAwNo3Oh27ugCakCoVeuCrZkvHMaJgcYrIGCuFo6q0Pfk9rsZyriIqEa9AtiUOtViInVYdby7y71wcbl0AbbCZsTSqnSoVxm2tRkOsXV6+8X4SnwcmZbao3H+zfO1GBhQOLxJ4NQbzAa8IJh810rYARNLptgmsd4cYXVOSosTX azureuser"
+)
+
+// ScaleSets returns a slice of VirtualMachineScaleSets within a given resource group
+func ScaleSets(ctx context.Context, logger *logrus.Entry, ssc azureclient.VirtualMachineScaleSetsClient, resourceGroup string) ([]compute.VirtualMachineScaleSet, error) {
+	vmssPages, err := ssc.List(ctx, resourceGroup)
+	if err != nil {
+		return nil, err
+	}
+
+	var scaleSets []compute.VirtualMachineScaleSet
+	for vmssPages.NotDone() {
+		scaleSets = append(scaleSets, vmssPages.Values()...)
+		err = vmssPages.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return scaleSets, nil
+}
+
+// ScaleSetVMs returns a slice of VirtualMachineScaleSetVMs within a given scale set
+func ScaleSetVMs(ctx context.Context, logger *logrus.Entry, ssvmc azureclient.VirtualMachineScaleSetVMsClient, resourceGroup string, scaleSet string) ([]compute.VirtualMachineScaleSetVM, error) {
+	vmPages, err := ssvmc.List(ctx, resourceGroup, scaleSet, "", "", "")
+	if err != nil {
+		return nil, err
+	}
+
+	var vms []compute.VirtualMachineScaleSetVM
+	for vmPages.NotDone() {
+		vms = append(vms, vmPages.Values()...)
+		err = vmPages.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return vms, nil
+}
+
+// UpdateScaleSetsCapacityErrors returns a slice of the errors it encounters as it attempts to increment the capacity
+// (by one) for all scale sets within a given resource group
+func UpdateScaleSetsCapacityErrors(ctx context.Context, logger *logrus.Entry, ssc azureclient.VirtualMachineScaleSetsClient, ssvmc azureclient.VirtualMachineScaleSetVMsClient, resourceGroup string) []error {
```

Comment:
Errors is probably redundant as a suffix in these helpers.

Reply:
I think the 'Errors' keyword in those function names are necessary to point out that those functions, instead of retrieving various azure resources, are only involved in retrieving errors.